test/test_dht_storage.cpp

```diff
@@ -121,20 +120,21 @@ TORRENT_TEST(announce_peer)
 TORRENT_TEST(put_immutable_item)
 {
 	dht_settings sett = test_settings();
-	boost::scoped_ptr<dht_storage_interface> s(dht_default_storage_constructor(node_id(0), sett));
+	node_id id(0);
+	boost::scoped_ptr<dht_storage_interface> s(dht_default_storage_constructor(sett));
 	TEST_CHECK(s.get() != NULL);
 
 	entry item;
 	bool r = s->get_immutable_item(n4, item);
 	TEST_CHECK(!r);
 
-	s->put_immutable_item(n4, "123", 3, address::from_string("124.31.75.21"));
+	s->put_immutable_item(n4, id, "123", 3, address::from_string("124.31.75.21"));
```

Comment:
as pointed out, the ID here is not a property of the value we're putting. it's inherently a property of all the nodes we're running